config/IpCfg.hpp

```diff
@@ -0,0 +1,29 @@
+// ======================================================================
+// \title  IpCfg.hpp
+// \author mstarch
+// \brief  hpp file for SocketIpDriver component implementation class
+//
+// \copyright
+// Copyright 2009-2015, by the California Institute of Technology.
+// ALL RIGHTS RESERVED.  United States Government Sponsorship
+// acknowledged.
+//
+// ======================================================================
+
+#ifndef REF_IPCFG_HPP
+#define REF_IPCFG_HPP
+
+enum IpCfg {
+    SOCKET_SEND_TIMEOUT_SECONDS = 1,       // Seconds component of timeout
+    SOCKET_SEND_TIMEOUT_MICROSECONDS = 0,  // Milliseconds component of timeout
+    SOCKET_IP_SEND_FLAGS = 0,                 // send, sendto FLAGS argument
+    SOCKET_IP_RECV_FLAGS = 0,                 // recv FLAGS argument
+    SOCKET_MAX_ITERATIONS = 0xFFFF,        // Maximum retry iterations
```

Comment:
Do you think it might make sense to make this is a private constant, not a configuration option?

We don't set recv timeouts, so the only thing that can trigger a retry are interrupts, which we likely want to transparently retry.

While we do have a send timeout, those can't safely be re-tried internally and should likely be surfaced to the user, leaving the only thing that can safely require a retry an interrupt.

I don't think most users will want to tweak the number of times we retry after a signal interrupt, can we reduce the "user" configuration complexity by hiding this constant?

Reply:
Given we let users set timeouts, we should also allow them to set the max iterations.  I am cleaning up the descriptive comments here to make it clearer what each does.